### PR TITLE
Switch language formatting and parsing to use Lingua Franca

### PR DESCRIPTION
**Pull request by forslund**
## Description
This aims to replace the internal language parsing and formatting with the [lingua franca](https://github.com/MycroftAI/lingua-franca/) module. It still aims to be backwards compatible so some logic remains in core along with all python modules.

- It leaves all python intact but replaces every class and function defined under mycroft/util/lang with imports from lingua-franca.
- util/(format|parse).py still contains some logic but mainly refers to lingua-franca
- the data files in /mycroft/res has been completely removed.

## How to test
Run core make sure things behaves. Do some explorative testing.

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

**Comment by forslund** [comment]
The coverage will drop and is acceptable.

File: mycroft/util/format.py [hunk 1]
```diff
@@ -20,79 +20,38 @@
...
-from mycroft.util.lang.format_da import nice_number_da
-from mycroft.util.lang.format_da import nice_time_da
-from mycroft.util.lang.format_da import pronounce_number_da
+from lingua_franca.format import (NUMBER_TUPLE, DateTimeFormat, join_list,
+                                  date_time_format, expand_options,
+                                  _translate_word)
+
+from lingua_franca.lang import get_full_lang_code, get_primary_lang_code
+
+from lingua_franca.lang.format_en import *
```
Comment (chrisveilleux) [inline mycroft/util/format.py]: This pattern concerns me.  The wildcard imports are a Python anti pattern.  Also, this import list can get unwieldy as we add more language support.  Why not have nice_number, nice_time and pronounce_number functions that take a language parameter?

**Comment by forslund** [comment]
Yes it should but...until a major release we need to keep the core-api as is.

File: mycroft/util/format.py (hunk 1 above)
Comment (forslund) [reply mycroft/util/format.py]: This is exactly what the file you're in does.

It is kept like this for backwards compatibility so it exports the expected symbols. In future releases this file will probably remain just importing the language-agnostic functions from lingua-franca.

Comment (chrisveilleux) [reply mycroft/util/format.py]: Ok, please add a comment or TODO indicating this.

**Comment by ChanceNCounter** [comment]
Seems to revert something about extract_datetime that breaks the weather skill. To repro: "Hey Mycroft, what's the weather like?" `NoneType is not subscriptable`

**Comment by forslund** [comment]
Thanks for reporting @ChanceNCounter totally reproducible. will look into and fix.

**Comment by forslund** [comment]
Hmm, seems like the lingua-franca behavior is the intended behavior: https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/util/parse.py#L248

**Comment by forslund** [comment]
Likely something similar to this fix: https://github.com/MycroftAI/mycroft-core/pull/2300

**Pull request merged**